source/internal.d.ts

```diff
@@ -328,3 +356,73 @@ IsPrimitive<Object>
 ```
 */
 export type IsPrimitive<T> = [T] extends [Primitive] ? true : false;
+
+/**
+Returns the fixed-indexed part of the given array.
+
+@example
+```
+type A = [string, number, boolean, ...string[]];
+type B = FixedPartOfArray<A>;
+//=> [string, number, boolean]
+```
+*/
+export type FixedPartOfArray<T extends UnknownArray, Result extends UnknownArray = []> =
+	T extends unknown
+		? number extends T['length'] ?
+			T extends readonly [infer U, ...infer V]
+				? FixedPartOfArray<V, [...Result, U]>
+				: Result
+			: T
+		: never; // Should never happen
+
+/**
+Returns the non-fixed-indexed part of the given array.
+
+@example
+```
+type A = [string, number, boolean, ...string[]];
+type B = NonFixedPartOfArray<A>;
+//=> string[]
+```
+*/
+export type NonFixedPartOfArray<T extends UnknownArray> =
+	T extends unknown
+		? T extends readonly [...FixedPartOfArray<T>, ...infer U]
+			? U
+			: []
+		: never; // Should never happen
+
+/**
+Returns the minimum number of the given union numbers.
+
+Note: Just supports numbers from 0 to 999.
+
+@example
+```
+type A = MinNumbers<3 | 1 | 2>
+//=> 1
+```
+*/
+export type MinNumbers<N extends Number, T extends UnknownArray = []> =
```

Comment:
```suggestion
export type MinNumbers<N extends number, T extends UnknownArray = []> =
```

Reply:
Should `T` really be exposed? May make more sense to wrap it in an internal type so this parameter is not exposed.